### PR TITLE
refactor: codegen modularization

### PR DESCRIPTION
**Pull request by Shikanime**
Implementation for #96

### CODE REVIEW

File: src/app/editor/code-editor/editor-container/codegen/shared-codegen.service.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          rules.map(rule => indentationSymbol + rule).join('\n'),
+          '}'
+        ].join('\n');
+      } else {
+        return null;
+      }
+    }
+  }
+
+  openTemplateBalise(tag = 'div', attributes = []) {
```
Comment (manekinekko) [inline src/app/editor/code-editor/editor-container/codegen/shared-codegen.service.ts]:
```suggestion
   openTag(tag = 'div', attributes = []) {
```

File: src/app/editor/code-editor/editor-container/codegen/shared-codegen.service.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    return `<${tag}${attributes.length !== 0 ? ' ' + attributes.join(' ') : ''}>`;
+  }
+
+  closeTemplateBalise(tag = 'div') {
```
Comment (manekinekko) [inline src/app/editor/code-editor/editor-container/codegen/shared-codegen.service.ts]:
```suggestion
   closeTemplateTag(tag = 'div') {
```

**Comment by Shikanime** [comment]
All done 🎉🎉

**Comment by Shikanime** [comment]
Oh duplicated commit let me fix that

**Comment by Shikanime** [comment]
It's now ready !

**Pull request merged**